File: kotlin/src/step/Start.kt
package step

fun start(state: Int): Pair<Int, CounterLead> {
    print("Enter a starting number (or press Enter for default): ")
    val input = readlnOrNull()
    return when {
        input.isNullOrEmpty() -> {
            println("Using default starting number.")
            state to CounterLead.COUNT
        }

        input.toIntOrNull() != null -> input.toInt() to CounterLead.COUNT
        else -> {
            println("Invalid number, please try again.")
            state to CounterLead.START
        }
    }
}

File: php/step/start.php
<?php

require_once __DIR__ . '/../lead.php';

function start($state) {
    while (true) {
        echo "Enter a starting number (or press Enter for default 0): ";
        $userInput = trim(fgets(STDIN));

        if (empty($userInput)) {
            echo "Starting count from default: $state\n";
            return [$state, Lead::COUNT];
        }

        if (is_numeric($userInput)) {
            $state = (int)$userInput;
            echo "Starting count from: $state\n";
            return [$state, Lead::COUNT];
        } else {
            echo "Invalid input. Please enter a valid number or press Enter.\n";
        }
    }
}

